File: app/src/main/java/juniojsv/minimum/SettingsActivity.java
package juniojsv.minimum;

import android.content.SharedPreferences;
import android.os.Bundle;
import android.preference.PreferenceActivity;

public class SettingsActivity extends PreferenceActivity {
    public static boolean needRestart;
    private SharedPreferences checkBoxDarkTheme;
    private SharedPreferences.Editor settingEditor = MinimumActivity.settings.edit();
    private final boolean DARK_THEME_PREF = MinimumActivity.settings.getBoolean("dark_theme", false);

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        applySettings();
        super.onCreate(savedInstanceState);
        addPreferencesFromResource(R.xml.settings_activity);
        checkBoxDarkTheme = findPreference("prefs_dark_theme").getSharedPreferences();
    }

    @Override
    protected void onPause() {
        super.onPause();
        updateSettings();
    }

    private void updateSettings() {
        if (checkBoxDarkTheme.getBoolean("prefs_dark_theme", false) != DARK_THEME_PREF) {
            settingEditor.putBoolean("dark_theme", checkBoxDarkTheme.getBoolean("prefs_dark_theme", false));
            needRestart = true;
        }

        if (needRestart) {
            settingEditor.apply();
            recreate();
        }
    }

    private void applySettings() {
        if (MinimumActivity.settings.getBoolean("dark_theme", false)) {
            setTheme(R.style.AppThemeDark);
        }
    }
}

File: app/src/main/java/juniojsv/minimum/SettingsActivity.kt
package juniojsv.minimum

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import juniojsv.minimum.SettingsManager.Companion.KEY_DARK_MODE
import kotlinx.android.synthetic.main.settings_activity.*

class SettingsActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        SettingsManager(this).also { settings ->
            if (settings.getBoolean(KEY_DARK_MODE)) setTheme(R.style.dark_mode)
            super.onCreate(savedInstanceState)

            setContentView(R.layout.settings_activity).apply {
                title = getString(R.string.settings_shortcut)
                dark_mode.isChecked = settings.getBoolean(KEY_DARK_MODE)
                version.text = String.format("%s %s", getString(R.string.version_title), BuildConfig.VERSION_NAME)
            }

            dark_mode.setOnCheckedChangeListener { _, isChecked ->
                settings.putBoolean(KEY_DARK_MODE, isChecked)
            }
        }
    }
}
